components/spin/index.tsx

```diff
@@ -116,6 +120,7 @@ class Spin extends React.Component<SpinProps, SpinState> {
   debouncifyUpdateSpinning = (props?: SpinProps) => {
     const { delay } = props || this.props;
     if (delay) {
+      this.cancelExistingSpin();
```

Comment:
Unsure if this should be outside the if and cancelled in all cases.  Can someone advise?